### PR TITLE
[Tooling] Dependabot groups

### LINKED ISSUE #9474: ✨ Dependabot merge groups
## ✨ Feature

It would be amazing if we could implement merge groups for dependabot.

## 🕵️ Details

A large reason dependabot merging takes so long is the fact that a lot of our dependencies are grouped so all have version updates at the same time (even when only one package has changed). Merging them separately causes merge conflicts meaning they need to be rebased and re-run our CI pipeline. This also has a domino effect significantly slowing the entire process down.

Dependabot has released [merge groups](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups) allowing us to merge dependency bumps to similar packages together. I think it would be a great help to implement this feature.

> [!NOTE]
> This may have some growing pains while we get the implementation correct 😢 

## 🙋‍♀️ Proposed Implementation

The largest contributor to the issue is storybook so we could start with that and expand on it once we are happy with it. But I have identified some other candidates we could also do if we want to dive right in 🏊 

- `@storybook/*`
- `@graphql-codegen/*`
- `@tiptap/*`
- `@urql/*`
- `@types/*` (maybe?) 🤷‍♀️ 
- `@formatjs/*`
- More?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Merge groups have been created
- [ ] Confirmed they work

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #9474 

## 👋 Introduction

Add merge groups.
Not sure how to confirm it works? Other than merge and wait until there are enough dependencies to update then trigger it.

Storybook, graphql, tiptap, and types were what I picked as they have multiples. Storybook I allowed major since that would probably have to be merged together anyway for it to not break. 

## 🧪 Testing

1. ?

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -13,6 +13,32 @@ updates:
       interval: "weekly"
       day: "thursday"
     open-pull-requests-limit: 50
+    groups:
+      storybook:
+        patterns:
+          - "@storybook*"
```
Comment (esizer) [inline .github/dependabot.yml:19]: I think we will also need `storybook` here for the main package that is not namespaced behind the org

**Review by esizer — CHANGES_REQUESTED** [review]
I think this is what we are looking for. Guess we just have to wait until one of these dependencies needs an update 😛 

Just one small suggestion for storybook.

File: .github/dependabot.yml (hunk 1 above)
Comment (vd1992) [reply .github/dependabot.yml:19]: Yes, yes you are right
https://github.com/GCTC-NTGC/gc-digital-talent/pull/9481/commits/954a1e0ce425c1a36b99b7edc26d26f2c0481811

**Review by esizer — APPROVED** [review]
LGTM! 👍 Now we pray 🙏 😅

**Pull request merged**